### PR TITLE
[proposal]When bitemporal_at exists inside the nest, the specified date was not prioritized, so the date of the inner bitemporal_at is now prioritized.

### PR DESCRIPTION
**Pull request by bonobono555**
# Summary
I think it's unnatural that the given date is not given priority when `bitemporal_at` exists inside the nest, and I want to give priority to the date of the inner `bitemporal_at`

reason
`valid_at` and `transaction_at` behave to give priority to the inner date, but `bitemporal_at` does not give priority to the inner date, so I thought it would be more natural to match `valid_at` and `transaction_at`.

ex. inner `valid_at` takes precedence
```
ActiveRecord::Bitemporal.valid_at("2020/01/01") do
  puts Employee.valid_at("1999/01/01").to_sql
end
# => SELECT "employees".*
#      FROM "employees"
#     WHERE "employees"."transaction_from" <= '2023-02-20 05:21:24'   <- current time
#       AND "employees"."transaction_to"   >  '2023-02-20 05:21:24'   <- current time
#       AND "employees"."valid_from"       <= '1999-01-01 00:00:00'
#       AND "employees"."valid_to"         >  '1999-01-01 00:00:00'
```

### bitemporal_at before
`bitemporal_at` does not take precedence
```
ActiveRecord::Bitemporal.valid_at("2020/01/01") do
  puts Employee.bitemporal_at("1999/01/01").to_sql
end
# => SELECT "employees".*
#      FROM "employees"
#     WHERE "employees"."transaction_from" <= '1999-01-01 00:00:00'
#       AND "employees"."transaction_to"   >  '1999-01-01 00:00:00'
#       AND "employees"."valid_from"       <= '2020-01-01 00:00:00'
#       AND "employees"."valid_to"         >  '2020-01-01 00:00:00'
```

### bitemporal_at after
`bitemporal_at` takes precedence
```
ActiveRecord::Bitemporal.valid_at("2020/01/01") do
  puts Employee.bitemporal_at("1999/01/01").to_sql
end
# => SELECT "employees".*
#      FROM "employees"
#     WHERE "employees"."transaction_from" <= '1999-01-01 00:00:00'
#       AND "employees"."transaction_to"   >  '1999-01-01 00:00:00'
#       AND "employees"."valid_from"       <= '1999-01-01 00:00:00'
#       AND "employees"."valid_to"         >  '1999-01-01 00:00:00'
```

### CODE REVIEW

**Review by osyo-manga — APPROVED** [review]
Great! :)

File: lib/activerecord-bitemporal/scope.rb [hunk 1]
```diff
@@ -407,15 +407,14 @@ def _#{column}_#{op}(datetime,**)
...
         relation = self
 
         if !ActiveRecord::Bitemporal.ignore_transaction_datetime?
-          relation = relation.transaction_at(ActiveRecord::Bitemporal.transaction_datetime || datetime)
+          relation = relation.transaction_at(datetime || ActiveRecord::Bitemporal.transaction_datetime || Time.current)
         end
 
         if !ActiveRecord::Bitemporal.ignore_valid_datetime?
-          relation = relation.valid_at(ActiveRecord::Bitemporal.valid_datetime || datetime)
+          relation = relation.valid_at(datetime || ActiveRecord::Bitemporal.valid_datetime || Time.current)
```
Comment (wata727) [inline lib/activerecord-bitemporal/scope.rb:417]: Calling `Time.current` on valid time and transaction time respectively can give different results in milliseconds. Is it better to use the same time? e.g.

```ruby
current_time = Time.current

if !ActiveRecord::Bitemporal.ignore_transaction_datetime?
  relation = relation.transaction_at(datetime || ActiveRecord::Bitemporal.transaction_datetime || current_time)
end

if !ActiveRecord::Bitemporal.ignore_valid_datetime?
  relation = relation.valid_at(datetime || ActiveRecord::Bitemporal.valid_datetime || current_time)
end
```

Comment (bonobono555) [reply lib/activerecord-bitemporal/scope.rb]: @wata727 
thank you.
Fixed as suggested.

**Review by wata727 — APPROVED** [review]
👏

**Pull request merged**